### ISSUE TITLE
[v2.26] az cli / acr purge crash: TypeError: 'NoneType' object is not callable

### ISSUE THREAD

**Issue by nulltoken** [microsoft/azure-pipelines-tasks#15036]
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: AzureCLI

list here (V# not needed):  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

## Environment
- Server - Azure Pipelines or TFS on-premises? Azure Pipelines

- Agent - Hosted or Private: Hosted

## Issue Description

This Azure Piepeline task was working fine in 2.25

```
      - task: AzureCLI@2
        displayName: Purge ancient Docker images
        inputs:
          azureSubscription: ${{ variables.serviceConnection }}
          scriptType: pscore
          scriptLocation: inlineScript
          inlineScript: |
            az acr run --cmd "acr purge --filter '[REDACTED]:.*' --ago $(daysToKeepImages.howMany)d --untagged" --registry $(registryName) /dev/null
```

Since the update to 2.26, it now crashes with the following output

```
ERROR: The command failed with an unexpected error. Here is the traceback:
ERROR: 'NoneType' object is not callable
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 657, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 691, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/acr/run.py", line 93, in acr_run
    variant=platform_variant
TypeError: 'NoneType' object is not callable
To open an issue, please run: 'az feedback'
```

Initially (wrongly) opened in https://github.com/MicrosoftDocs/azure-docs-cli/issues/2659

**Comment by tiwarishub** [comment]
hi @nulltoken, thanks for reaching out. Please share the debug logs and are you able to run the same command on the Azure portal

**Comment by dansaltmer** [comment]
Hi, We're also running into this on our live pipeline builds, was working yesterday, so I've done a little bit of digging. Our scenario is running a windows-2019 devops pipeline to process some legacy resource files, which then uses `az acr build` to run the actual linux image build on the ACR.

The issue is present in 2.26.0 of the CLI, and has been hotfixed in 2.26.1, details of this at:
https://github.com/Azure/azure-cli/issues/18860
https://docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?tabs=azure-cli#july-14-2021

The current version of the documentation for the windows-2019 image is slightly incorrect, it shows version 2.25.0 of the cli, but the image set up looks to just pull in the latest version of the CLI. According to https://github.com/actions/virtual-environments this image was built on the 11th and in the process of rollout, which has pulled in version 2.26.0 with the azure cli bug, this is confirmed in the version output during the pipelines themselves.

**Comment by nulltoken** [comment]
@dansaltmer Thanks a lot for the heads up! I can confirm that 2.26.1 fixes this.

Fixed by Azure/azure-cli#18860
